### PR TITLE
tests: do not use `assert!(r.is_ok())` in tests

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

### What is changed and how it works?

Issue Number: Ref #13008

What's Changed:

I'm trying to fix a flaky test, but the error isn't printed due to the use of `assert!(r.is_ok())`.

```commit-message
None
```

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by BusyJay** [comment]
Cool, can it be a lint?

**Comment by tabokie** [comment]
Disallowing `is_ok` entirely is an overkill, and clippy doesn't support configuring for complex things like `assert!(..)`. I filed an issue to clippy about this specific lint.

File: components/tikv_util/src/config.rs [hunk 1]
```diff
@@ -1944,7 +1943,7 @@ mod tests {
         let ret = check_data_dir_empty(tmp_path.to_str().unwrap(), "txt");
         assert!(ret.is_err());
         let ret = check_data_dir_empty(tmp_path.to_str().unwrap(), "xt");
-        assert!(ret.is_ok());
+        ret.unwrap();
```
Comment (BusyJay) [inline components/tikv_util/src/config.rs:1946]: Why not merge with the previous line?

File: components/tikv_util/src/yatp_pool/future_pool.rs [hunk 2]
```diff
@@ -472,10 +472,10 @@ mod tests {
         // full
         assert!(spawn_long_time_future(&read_pool, 8, 100).is_err());
 
-        assert!(rx.recv().is_ok());
-        assert!(rx.recv().is_ok());
-        assert!(rx.recv().is_ok());
-        assert!(rx.recv().is_ok());
+        rx.recv().unwrap().unwrap();
```
Comment (BusyJay) [inline components/tikv_util/src/yatp_pool/future_pool.rs:475]: Why unwrap twice?

Comment (tabokie) [reply components/tikv_util/src/yatp_pool/future_pool.rs:475]: I think it should check the task succeed.

File: components/tikv_util/src/config.rs (hunk 1 above)
Comment (tabokie) [reply components/tikv_util/src/config.rs:1946]: I didn't bother to type really.

Comment (BusyJay) [reply components/tikv_util/src/config.rs:1946]: Any updates?

**Review by BusyJay — APPROVED** [review]

**Review by MyonKeminta — APPROVED** [review]

**Pull request merged**